pgjdbc/src/main/java/org/postgresql/jdbc/PgCallableStatement.java

```diff
@@ -691,7 +696,120 @@ public Reader getCharacterStream(String parameterName) throws SQLException {
   }
 
   public <T> T getObject(int parameterIndex, Class<T> type) throws SQLException {
-    if (type == ResultSet.class) {
+    if (type == null) {
+      throw new SQLException("type is null");
+    }
+    checkIndex(parameterIndex);
+    int sqlType = functionReturnType[parameterIndex - 1];
+    if (type == BigDecimal.class) {
+      if (sqlType == Types.NUMERIC || sqlType == Types.DECIMAL) {
+        return type.cast(getBigDecimal(parameterIndex));
+      } else {
+        throw new SQLException("conversion to " + type + " from " + sqlType + " not supported");
+      }
+    } else if (type == String.class) {
+      if (sqlType == Types.CHAR || sqlType == Types.VARCHAR) {
+        return type.cast(getString(parameterIndex));
+      } else {
+        throw new SQLException("conversion to " + type + " from " + sqlType + " not supported");
+      }
+    } else if (type == Boolean.class) {
+      if (sqlType == Types.BOOLEAN || sqlType == Types.BIT) {
```

Comment:
Unfortunately it already does for every call. Let me explain:

- [PgResultSet#internalGetObject](https://github.com/pgjdbc/pgjdbc/blob/master/pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java#L175) already maps `BIT` to `boolean` no matter the length.
- [PgResultSet#getObject(int)](https://github.com/pgjdbc/pgjdbc/blob/master/pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java#L2589) calls `#internalGetObject`
- ultimately [BooleanTypeUtil#fromString](https://github.com/pgjdbc/pgjdbc/blob/master/pgjdbc/src/main/java/org/postgresql/jdbc/BooleanTypeUtil.java#L54) is called which fails 

There are several unpleasant consequences:
1. Because [PgCallableStatement](https://github.com/pgjdbc/pgjdbc/blob/master/pgjdbc/src/main/java/org/postgresql/jdbc/PgCallableStatement.java#L123) always calls `getObject(int)` calling a stored procedure that returns `BIT(>1)` always fails, this is already the case now.
1. Calling `(Pg)ResultSet#getObject(int)` on a column with `BIT(>1)` always fails, this is already the case now.

I agree this is an issue but IMHO this is a separate issue. Several questions arise like to what should `getObject(int)` map to for a `BIT` column? Should it be independent of the length? There is also a certain tension with the JDBC specification as Table B-6 suggests mapping `BIT` to `boolean`. There is also a possible overlap with #895. I would like to track this is a different issue, let me know if this is ok for you.

I will certainly add a unit test.